Source/WebCore/svg/animation/SVGSMILElement.cpp

```diff
@@ -119,8 +119,9 @@ bool ConditionEventListener::operator==(const EventListener& listener) const
 
 void ConditionEventListener::handleEvent(ScriptExecutionContext&, Event&)
 {
-    if (RefPtr animation = m_animation.get())
-        animation->handleConditionEvent(m_condition);
+    if (!m_animation)
+        return;
+    m_animation->addInstanceTime(m_condition->m_beginOrEnd, m_animation->elapsed() + m_condition->m_offset);
```

Comment:
Not good to remove the object lifetime change where we keep the animation in a local `RefPtr`.

Reply:
Done - reintroduced it in https://github.com/WebKit/WebKit/commit/f397d57751ebb138f81c2c693fdbd4bab66218b7.